lib/charms/vault_k8s/v0/vault_kv.py

```diff
@@ -519,15 +533,19 @@ def __init__(
         charm: ops.CharmBase,
         relation_name: str,
         mount_suffix: str,
+        refresh_events: Optional[List] = None,
```

Comment:
The fact that the default is None is somewhat misleading as if it will be set to `config_changed` in this case.  If we want it to be config changed, why not place it here and explain why it's the default in the docstring (which I realise we don't have at the moment)

Reply:
in my opinion this argument can be deleted and only config changed should be observed in the library, but if we really want it to stay I agree that it should default to config changed.

Reply:
@gruyaume Do you have any opinions on completely removing it as @Kayra1 suggests? I'd keep it at least for the flexibility it gives but on the other hand I don't have a use case in mind where users would chose a different event.

Reply:
If we don't have a use case for it, let's not add this unneeded flexibility. And if we keep the config_changed observing, let's mention somewhere why we do it because it's not straightforward why we care about this event.

Reply:
Done.